parser-typechecker/src/Unison/Codebase/FileCodebase.hs

```diff
@@ -128,10 +127,8 @@ initCodebaseAndExit mdir = do
   exitSuccess
 
 absDir :: FilePath -> IO FilePath
-absDir dir 
-  | dir == "." = makeAbsolute dir
-  | dir == ".." = takeDirectory <$> makeAbsolute "."
-  | otherwise = pure dir
+absDir dir =
+  if dir == "." || dir == ".." then canonicalizePath dir else pure dir
```

Comment:
Oh, I meant... `absDir` could just be replaced with `canonicalizaPath`, right?

Reply:
The behaviour is slightly different (for the better) in that now when the user supplies a path like "blah" and gets an error, the message shows the full path, which is the behaviour for `.`/`..`. 
So uniform and matching errors for all instances.

Reply:
Sorry, I didn't quite understand.  Is there something special that you wanted to do when displaying a path that _ends_ in `.` or `..`, but don't want to do if there's a `/./` or `/../` in the middle?

Maybe a few examples of what you had in mind?

Reply:
I updated the examples of usage in my previous comment, but omitted to mention it explicitly. (https://github.com/unisonweb/unison/pull/1325#issuecomment-596184930)

When a user supplies a path and as long as `canonicalyzePath` can translate it, the error produced ("run init, cannot init exists") will contain the translated path exactly the way unison/haskell/os see it internally.

I have attached examples showing how the errors and output work, I am not trying to do permutation testing with `/.`/ or `/../` as ultimately that would be testing the implementation of `canonicalyzePath`.